src/OpenTelemetry/Trace/Span.cs

```diff
@@ -99,11 +99,11 @@ public sealed class Span : ISpan
 
                 if (links != null)
                 {
-                    this.links = new EvictingQueue<Link>(this.tracerConfiguration.MaxNumberOfLinks);
+                    this.links = new List<Link>(this.tracerConfiguration.MaxNumberOfLinks);
 
-                    foreach (var link in links)
+                    foreach (var link in links.Reverse().Take(this.tracerConfiguration.MaxNumberOfLinks))
```

Comment:
cmon, Reverse is a cheat and should be prohibited ;)

Reply:
i mean you could always do decreasing index without allocating another array

Reply:
Right! I thought so. :D
Considering `.ToList()` on a list doesn't allocate, can I get around with this then?